x-pack/plugins/security_solution/public/users/pages/navigation/user_risk_tab_body.tsx

```diff
@@ -0,0 +1,130 @@
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { EuiFlexGroup, EuiFlexItem } from '@elastic/eui';
+import React, { useCallback, useMemo } from 'react';
+import styled from 'styled-components';
+
+import * as i18n from '../translations';
+
+import { useQueryInspector } from '../../../common/components/page/manage_query';
+import { RiskScoreOverTime } from '../../../common/components/risk_score_over_time';
+import { TopRiskScoreContributors } from '../../../common/components/top_risk_score_contributors';
+import { useQueryToggle } from '../../../common/containers/query_toggle';
+import { UserRiskScoreQueryId, useUserRiskScore } from '../../../risk_score/containers';
+import { buildUserNamesFilter } from '../../../../common/search_strategy';
+import { UsersComponentsQueryProps } from './types';
+import { UserRiskInformationButtonEmpty } from '../../components/user_risk_information';
+
+const QUERY_ID = UserRiskScoreQueryId.USER_DETAILS_RISK_SCORE;
+
+const StyledEuiFlexGroup = styled(EuiFlexGroup)`
+  margin-top: ${({ theme }) => theme.eui.paddingSizes.l};
+`;
+
+const UserRiskTabBodyComponent: React.FC<
+  Pick<UsersComponentsQueryProps, 'startDate' | 'endDate' | 'setQuery' | 'deleteQuery'> & {
+    userName: string;
+  }
+> = ({ userName, startDate, endDate, setQuery, deleteQuery }) => {
+  const timerange = useMemo(
+    () => ({
+      from: startDate,
+      to: endDate,
+    }),
+    [startDate, endDate]
+  );
+
+  const { toggleStatus: overTimeToggleStatus, setToggleStatus: setOverTimeToggleStatus } =
+    useQueryToggle(`${QUERY_ID} overTime`);
+  const { toggleStatus: contributorsToggleStatus, setToggleStatus: setContributorsToggleStatus } =
+    useQueryToggle(`${QUERY_ID} contributors`);
+
+  const [loading, { data, refetch, inspect }] = useUserRiskScore({
+    filterQuery: userName ? buildUserNamesFilter([userName]) : undefined,
+    onlyLatest: false,
+    skip: !overTimeToggleStatus && !contributorsToggleStatus,
+    timerange,
+  });
+
+  useQueryInspector({
+    queryId: QUERY_ID,
+    loading,
+    refetch,
+    setQuery,
+    deleteQuery,
+    inspect,
+  });
+
+  const toggleContributorsQuery = useCallback(
+    (status: boolean) => {
+      setContributorsToggleStatus(status);
+    },
+    [setContributorsToggleStatus]
+  );
+
+  const toggleOverTimeQuery = useCallback(
+    (status: boolean) => {
+      setOverTimeToggleStatus(status);
+    },
+    [setOverTimeToggleStatus]
+  );
+
+  const rules = data && data.length > 0 ? data[data.length - 1].risk_stats.rule_risks : [];
+
+  return (
+    <>
+      <EuiFlexGroup direction="row">
+        <EuiFlexItem grow={2}>
+          <RiskScoreOverTime
+            from={startDate}
+            to={endDate}
+            loading={loading}
+            riskScore={data}
+            queryId={QUERY_ID}
```

Comment:
shouldn't this be the same id that's passed to `useQueryToggle`?

Reply:
On this page, we have two visualizations using the same API call / QUERY_ID  🤯  

To be able to toggle the visualization individually, I got creative and derived the ids for `useQueryToggle` from `QUERY_ID`. 

```tsx
    useQueryToggle(`${QUERY_ID} overTime`);
    useQueryToggle(`${QUERY_ID} contributors`);
```